contracts/MarketOracle.sol

```diff
@@ -97,9 +151,10 @@ contract MedianOracle is Ownable, IOracle {
         external
         onlyOwner
     {
-        require(providerReports[provider].timestamp == 0);
+        require(providerReports[provider][0].timestamp == 0 && providerReports[provider][1].timestamp == 0);
```

Comment:
Side question: why is addProvider `external`, but removeProvider `public`?

Reply:
Good observation switching to external. I think RemoveProvider was called from inside in a pervious version of the code and was inherited accidentally.